zio-http/src/main/scala/zhttp/http/middleware/HttpMiddleware.scala

```diff
@@ -255,26 +291,27 @@ object HttpMiddleware {
   def removeHeader(name: String): HttpMiddleware[Any, Nothing] =
     patch((_, _) => Patch.removeHeaders(List(name)))
 
+  private def equalsIgnoreCase(a: Char, b: Char) = a == b || toLowerCase(a) == toLowerCase(b)
+  private def contentEqualsIgnoreCase(a: CharSequence, b: CharSequence): Boolean = {
+    if (a == b)
+      true
+    else if (a.length() != b.length())
+      false
+    else if (a.isInstanceOf[AsciiString]) {
+      a.asInstanceOf[AsciiString].contentEqualsIgnoreCase(b)
+    } else if (b.isInstanceOf[AsciiString]) {
+      b.asInstanceOf[AsciiString].contentEqualsIgnoreCase(a)
+    } else {
+      (0 until a.length()).forall(i => equalsIgnoreCase(a.charAt(i), b.charAt(i)))
+    }
+  }
```

Comment:
These functions are also defined inside of `HeaderExtensions` can we move them to a common place.